enhancements/configurable-dev-env.md

```diff
@@ -0,0 +1,409 @@
+Links: 6 point-to-point, 1 broadcast network
+
+Edge Leaves
+-----------
+leafA (matched pattern: leaf[AB])
+  Role: edge-leaf
+  Router ID: 10.0.1.1
+  VTEP IP: 10.0.1.1
+  Interfaces:
+    eth1 -> spine (192.168.0.1/31, fd00::1/127)
+    ethred -> host-red (VRF: red)
+    ethblue -> host-blue (VRF: blue)
+  VRFs:
+    red (VNI: 100, interfaces: [ethred])
+    blue (VNI: 101, interfaces: [ethblue])
+  BGP: AS 65001, 1 peer (spine), EVPN enabled
+
+leafB (matched pattern: leaf[AB])
+  Role: edge-leaf
+  Router ID: 10.0.1.2
+  VTEP IP: 10.0.1.2
+  Interfaces:
+    eth1 -> spine (192.168.0.3/31, fd00::3/127)
+    ethred -> host-red (VRF: red)
+    ethblue -> host-blue (VRF: blue)
+  VRFs:
+    red (VNI: 100, interfaces: [ethred])
+    blue (VNI: 101, interfaces: [ethblue])
+  BGP: AS 65001, 1 peer (spine), EVPN enabled
+
+Transit Nodes
+-------------
+spine (matched pattern: spine)
+  Role: transit
+  Router ID: 10.0.0.1
+  Interfaces:
+    eth1 -> leafA (192.168.0.0/31, fd00::0/127)
+    eth2 -> leafB (192.168.0.2/31, fd00::2/127)
+    eth3 -> leafkind-switch (192.168.1.1/24, fd00:1::1/64)
+  BGP: AS 65000, 3 peers (leafA, leafB, leafkind), EVPN route reflector
+
+Configuration applied successfully.
+```
+
+### Persistence
+
+The summary can be regenerated at any time using:
+
+```bash
+clab-config summary --state topology-state.json
+```
+
+This allows users to review the configuration without re-running the full generation process.
+
+## IP Assignment Strategy
+
+### Point-to-Point Links
+
+The system automatically allocates IP addresses for all point-to-point links between routers:
+
+- **IPv4**: `/31` subnets (2 usable addresses per link)
+- **IPv6**: `/127` subnets (2 usable addresses per link)
+
+Allocation occurs from configurable base ranges (e.g., `192.168.0.0/16` for IPv4, `fd00::/48` for IPv6).
+
+### Switch/Broadcast Networks
+
+Switches (bridge nodes in containerlab) represent broadcast domains and require different handling:
+
+- **IPv4**: `/24` subnet (254 usable addresses)
+- **IPv6**: `/64` subnet
+
+All interfaces connected to the same switch receive IPs from the same subnet. This supports scenarios like the `leafkind-switch` which connects multiple nodes (leafkind, pe-kind-control-plane, pe-kind-worker).
+
+### Special Allocations
+
+Additional IP ranges may be reserved for:
+
+- **VTEP addresses**: Loopback IPs for VXLAN tunnel endpoints
+- **Router IDs**: BGP router identifiers (typically IPv4 addresses)
+- **Management**: Out-of-band management interfaces (if needed)
+
+## Automatic Property Assignment
+
+The system automatically assigns variable properties that would otherwise require manual specification for each node:
+
+### VTEP IPs (for edge leaves)
+- Allocated from a dedicated loopback range (e.g., `10.0.1.0/24`)
+- One unique IP per edge leaf node
+- Used as the source IP for VXLAN tunnels
+
+### MAC Addresses
+- Generated deterministically or randomly for VXLAN interfaces
```

Comment:
we can't have random, can we ? We need to implement a distributed GW on the leafkind switches at least, so the workloads all have the same GW MAC + IP. 

Maybe that should be a configuration a new configuration - i.e. you get to define a distributed anycast GW for your edge leaf.